apps/explorer/src/ui/IconButton.tsx

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { ReactComponent as CloseIcon } from './icons/x.svg';
+import { ButtonOrLink, type ButtonOrLinkProps } from './utils/ButtonOrLink';
+
+import type { FC } from 'react';
+
+type IconType = 'x';
+const iconTypeToIcon: Record<IconType, FC> = {
+    x: CloseIcon,
+};
+
+export interface IconButtonProps extends Omit<ButtonOrLinkProps, 'children'> {
```

Comment:
I'd add a required `aria-label` prop (which already exists on `ButtonOrLinkProps`, but optionally), to ensure all buttons are correctly labelled.